engine/lib/phx/src/math/ray.rs

```diff
@@ -3,19 +3,19 @@ use super::*;
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub struct Ray {
-    pub p: Vec3,
+    pub p: Position,
     pub dir: Vec3,
```

Comment:
f64 for dir, min and max?

Reply:
I think it's reasonable to leave normalised directions as Vec3, we don't need to change everything to f64.

I agree about min and max though, I'll get those updated.

Reply:
> I think it's reasonable to leave normalised directions as Vec3, we don't need to change everything to f64.

Double precision could be important if we 'shoot' ray on a longer distance but for now it's probably ok.

Reply:
Hm, that's a good point. I was imagining that we would do something like `self.dir.as_dvec() * ray_distance` but then it might as well be a DVec3.